From c170c41062518a5f5f0c4c16feda6cdc07010c75 Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Wed, 11 Dec 2024 18:43:47 +0200
Subject: [PATCH] testsuite.at: Update testsuite config for MSVC

* tests/testsuite.at: Make symlink tool mlink and c++ compiler equal
to the c compiler for MSVC.
---
diff --git a/tests/testsuite.at b/tests/testsuite.at
index bacc72ce..ea3b80e3 100644
--- a/tests/testsuite.at
+++ b/tests/testsuite.at
@@ -72,6 +72,17 @@ unset MFLAGS MAKEFLAGS MAKELEVEL __MKLVL__ MAKE_JOBS_FIFO
 case $host_os in
 aix*) umask o-rwx ;;
 esac
+
+# Use mklink when symlinking for MSVC
+if $LIBTOOL --config | $EGREP '^nm_interface="MS dumpbin"' >/dev/null; then
+  case $host_os in
+  mingw*) # MSYS
+    LN_S="cmd //c mklink" ;;
+  *)
+    LN_S="cmd /c mklink" ;;
+  esac
+  CXX=$CC
+fi
 m4_divert_pop([PREPARE_TESTS])dnl
 
 
--
2.45.2

